src/main/java/bowling/domain/Frame.java

```diff
@@ -1,62 +1,79 @@
 package bowling.domain;
 
-import bowling.dto.FrameDto;
-
 import java.util.ArrayList;
 import java.util.List;
 
-class Frame {
+public class Frame {
     private static final int SHOT_LIMIT = 2;
 
     private final ShotScores shotScores;
-    private final boolean hasBonus;
+    private final boolean islast;
 
-    protected Frame(List<ShotScore> shotScores, boolean hasBonus) {
-        this.shotScores = new ShotScores(shotScores);
-        this.hasBonus = hasBonus;
+    protected Frame(List<ShotScore> shotScores, boolean islast) {
+        this.shotScores = ShotScores.of(shotScores);
+        this.islast = islast;
     }
 
     static Frame init() {
         return new Frame(new ArrayList<>(), false);
     }
 
     Frame next(int shot) {
-        Frame frame = new Frame(new ArrayList<>(), false);
-        frame.shot(shot);
-        return frame;
+        Frame nextFrame = new Frame(new ArrayList<>(), false);
+        nextFrame.shot(shotScores.getNext(shot));
+        return nextFrame;
     }
 
     Frame last(int shot) {
-        Frame frame = new Frame(new ArrayList<>(), true);
-        frame.shot(shot);
-        return frame;
+        Frame nextFrame = new Frame(new ArrayList<>(), true);
+        nextFrame.shot(shotScores.getNext(shot));
+        return nextFrame;
     }
 
     void shot(int shot) {
-        if (shotScores.isSize(getShotLimit())) {
-            throw new IllegalStateException(String.format("shot NormalFrame fail. cannot shot over %d times", getShotLimit()));
+        shot(shotScores.getNext(shot));
+    }
+
+    private void shot(ShotScore shot) {
+        int shotLimit = getShotLimit();
+        if (shotScores.isSize(shotLimit)) {
+            throw new IllegalStateException(String.format("shot Frame fail. cannot shot over %d times", shotLimit));
         }
 
-        shotScores.add(shot, hasBonus);
+        shotScores.add(shot);
     }
 
     boolean isFrameSet() {
         return shotScores.isSize(getShotLimit()) ||
                 (cannotShooting());
     }
 
+    private int getShotLimit() {
+        return islast ? SHOT_LIMIT + 1 : SHOT_LIMIT;
+    }
+
     private boolean cannotShooting() {
-        if (hasBonus) {
-            return shotScores.isSize(SHOT_LIMIT) && !shotScores.isClear();
+        if (islast) {
+            return shotScores.isSize(SHOT_LIMIT) && !shotScores.hasClear();
         }
-        return shotScores.isClear();
+        return shotScores.hasClear();
     }
 
-    FrameDto getDto() {
-        return new FrameDto(shotScores.getDtoList());
+    Integer getFrameScore() {
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
    public Integer getFrameScore() {
```

Reply:
아이고;;;... 접근제한자 실수를 했군요..ㅠㅜ

Reply:
🆗